roles/kubernetes/node/templates/kubelet.j2

```diff
@@ -6,8 +6,9 @@ KUBELET_ADDRESS="--address={{ ip | default("0.0.0.0") }}"
 # The port for the info server to serve on
 # KUBELET_PORT="--port=10250"
 # You may leave this blank to use the actual hostname
-KUBELET_HOSTNAME="--hostname-override={{ ansible_hostname }}"
-
+{% if kube_override_hostname %}
```

Comment:
You can't "unset" a variable in ansible. Therefore you have to set this variable to false to disable overriding the hostname. Is that your goal?